karma/test-services.coffee

```diff
@@ -1,30 +1,410 @@
+  )
+
+  #=======================================================================
+  # Tests for CaseService
+  #=======================================================================
+  describe('CaseService', () ->
+    CaseService = null
+
+    beforeEach(inject((_CaseService_) ->
+      CaseService = _CaseService_
+
+      test.case1 = {
+        id: 501,
+        summary: "Got tea?",
+        labels: [test.tea],
+        assignee: test.moh,
+        opened_on: utcdate(2016, 5, 27, 11, 0, 0, 0),
+        is_closed: false
+      }
+    ))
+    
+    describe('addNote', () ->
+      it('posts to note endpoint', () ->
+        $httpBackend.expectPOST('/case/note/501/').respond('')
+        CaseService.addNote(test.case1, "Hello there")
+        $httpBackend.flush()
+      )
+    )
+
+    describe('fetchNew', () ->
+      it('gets cases from search endpoint', () ->
+        $httpBackend.expectGET('/case/search/?folder=open').respond('{"results":[{"id":501,"opened_on":"2016-05-17T08:49:13.698864"}],"has_more":true}')
+        CaseService.fetchNew({folder: "open"}).then((data) ->
+          expect(data.results).toEqual([{id: 501, opened_on: utcdate(2016, 5, 17, 8, 49, 13, 698)}])
+        )
+        $httpBackend.flush()
+      )
+    )
+
+    describe('fetchOld', () ->
+      it('gets cases from search endpoint', () ->
+        $httpBackend.expectGET('/case/search/?folder=open').respond('{"results":[{"id":501,"opened_on":"2016-05-17T08:49:13.698864"}],"has_more":true}')
+        CaseService.fetchOld({folder: "open"}).then((data) ->
+          expect(data.results).toEqual([{id: 501, opened_on: utcdate(2016, 5, 17, 8, 49, 13, 698)}])
+          expect(data.hasMore).toEqual(true)
+        )
+        $httpBackend.flush()
+      )
+    )
+
+    describe('fetchTimeline', () ->
+      it('gets cases from timeline endpoint', () ->
+        $httpBackend.expectGET('/case/timeline/501/?after=2016-05-28T09:00:00.000Z').respond('{"results":[{"id":501,"time":"2016-05-17T08:49:13.698864","type":"A"}]}')
+        CaseService.fetchTimeline(test.case1, utcdate(2016, 5, 28, 9, 0, 0, 0)).then((data) ->
+          expect(data.results).toEqual([{
+            id: 501,
+            time: utcdate(2016, 5, 17, 8, 49, 13, 698),
+            type: 'A',
+            is_action: true,
+            is_message_in: false,
+            is_message_out: false
+          }])
+        )
+        $httpBackend.flush()
+      )
+    )
+
+    describe('fetchSingle', () ->
+      it('gets case from fetch endpoint', () ->
+        $httpBackend.expectGET('/case/fetch/501/').respond('{"id":501,"opened_on":"2016-05-17T08:49:13.698864"}')
+        CaseService.fetchSingle(501).then((caseObj) ->
+          expect(caseObj).toEqual({id: 501, opened_on: utcdate(2016, 5, 17, 8, 49, 13, 698)})
+        )
+        $httpBackend.flush()
+      )
+    )
+
+    describe('reassign', () ->
+      it('posts to reassign endpoint', () ->
+        $httpBackend.expectPOST('/case/reassign/501/').respond('')
```

Comment:
Currently these tests aren't checking the post body because the requests use `FormData` and that's pretty awkward to mock or verify. In a subsequent PR I'll switch these to use urlencoded form submission which will make it easier to test here and make param handling simpler in general.